src/Mono.Android/Microsoft.Android.Runtime/ManagedValueManager.cs

```diff
@@ -61,13 +66,17 @@ public override void CollectPeers ()
 		}
 		if (exceptions != null)
 			throw new AggregateException ("Exceptions while collecting peers.", exceptions);
+
+		GC.Collect ();
 	}
 
 	public override void AddPeer (IJavaPeerable value)
 	{
 		if (RegisteredInstances == null)
 			throw new ObjectDisposedException (nameof (ManagedValueManager));
 
+		WaitForGCBridgeProcessing ();
```

Comment:
Not sure what race this wait is meant to prevent. Even if we do the wait here, the code below could still race with a bridge collection.